File: PashmakAndFlowers.java
import java.util.Scanner;

public class PashmakAndFlowers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        long[] b = new long[n];
        
        long min = Long.MAX_VALUE;
        long max = Long.MIN_VALUE;
        
        for (int i = 0; i < n; i++) {
            b[i] = scanner.nextLong();
            if (b[i] < min) min = b[i];
            if (b[i] > max) max = b[i];
        }
        
        long diff = max - min;
        
        if (diff == 0) {
            long ways = (long) n * (n - 1) / 2;
            System.out.println(diff + " " + ways);
        } else {
            long countMin = 0, countMax = 0;
            for (int i = 0; i < n; i++) {
                if (b[i] == min) countMin++;
                if (b[i] == max) countMax++;
            }
            System.out.println(diff + " " + (countMin * countMax));
        }
        
        scanner.close();
    }
}

File: PashmakAndFlowers.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

int main() {
    int n;
    cin >> n;
    
    vector<long long> b(n);
    long long minVal = LLONG_MAX;
    long long maxVal = LLONG_MIN;
    
    for (int i = 0; i < n; i++) {
        cin >> b[i];
        if (b[i] < minVal) minVal = b[i];
        if (b[i] > maxVal) maxVal = b[i];
    }
    
    long long diff = maxVal - minVal;
    
    if (diff == 0) {
        long long ways = (long long)n * (n - 1) / 2;
        cout << diff << " " << ways << endl;
    } else {
        long long countMin = 0, countMax = 0;
        for (int i = 0; i < n; i++) {
            if (b[i] == minVal) countMin++;
            if (b[i] == maxVal) countMax++;
        }
        cout << diff << " " << (countMin * countMax) << endl;
    }
    
    return 0;
}